### DISCUSSION TITLE
Disable tracing of input and output for nested elements with in a chain.

### DISCUSSION THREAD

**Discussion by jaitd** [langfuse/langfuse#2706] [Support]
We are using the decorator API and the Lanchain integration. We are also self-hosting Langfuse, currently on `v2.63.1`.

We wanted to in specific cases NOT trace the input and output for our all the steps in our chain. We are doing this by wrapping the observe decorator in a factory which sets `capture_input` and `capture_output` to `False` when needed. This does work i.e. we don't see the input and output for the function that is decorated with this decorator. The problem is that the nested spans and Runnable still have their input and output available. We would like to disable tracing input and output for these as well.

Here is what our chain setup looks like:

```
def conditional_observe(func):
    @wraps(func)
    def wrapper(*args, **kwargs):
        condition = "something from the args"
        if condition:
            observed_func = observe(capture_input=False, capture_output=False)(func)
        else:
            observed_func = observe(capture_input=True, capture_output=True)(func)
        return observed_func(*args, **kwargs)
    return wrapper

@conditional_observe()
def conversation_chain(arg1: str, arg2: str, api_key: str, prompts):
    model = ChatOpenAI(
        model="gpt-4-turbo",
        temperature=0,
        max_tokens=1000,
        model_kwargs={"frequency_penalty": 2.0},
        api_key=api_key,
    )

    parallel_chain = {
        f"chain_{idx}": ChatPromptTemplate.from_messages(
            [
                ("system", prompt),
                ("human", "{user_input}"),
            ]
        )
        | model
        | StrOutputParser()
        for idx, prompt in enumerate(prompts)
    }

    synthesizer_chain = (
        ChatPromptTemplate.from_messages([("system", SYNTHESIZER_PROMPT)]) | model
    )

    chain = {
        "input1": itemgetter("input1"),
        "input2": itemgetter("input2"),
        "input3": itemgetter("input3"),
        "input4": itemgetter("input4"),
        "input5": itemgetter("input5"),
        "input6": parallel_chain
        | RunnableLambda(
            lambda input_dict: [
                value for key, value in input_dict.items() if key.startswith("chain_")
            ]
        ),
    } | synthesizer_chain

    return chain.invoke(
        {
            "input1": "...",
            "input2": "...",
            "input3": "...",
            "input4": "...",
            "input5": "...",
        },
        config={"callbacks": [langfuse_handler]}
    )
```

Is there a way in which we can disable the input and output values in tracing for all the nested spans/`Runnable`s in the chain?

**Comment by marcklingen** [comment]
Currently, you only have full control over this behavior when using the decorator or the low-level SDK. The Langchain integration automatically collects all inputs and outputs, but it cannot be configured in this level of detail.

We are open to contributions on this, as I believe it would be feasible to add additional configuration options to a Langchain class, which could then be picked up by the Langfuse callback handler to ignore the input/output.

**Reply by jaitd** [reply]
Thanks for the quick response, Marc.

I'll take look a the Langchain integration and see if I can cobble up a solution for it.

**Reply by marcklingen** [reply]
related:
- https://github.com/orgs/langfuse/discussions/2714
- https://github.com/orgs/langfuse/discussions/2718#discussioncomment-10125815
